kotlinx-coroutines-core/common/src/channels/BufferedChannel.kt

```diff
@@ -1583,6 +1583,7 @@ internal open class BufferedChannel<E>(
          * continuation. The [tryResumeHasNext] and [tryResumeHasNextOnClosedChannel]
          * function resume this continuation when the [hasNext] invocation should complete.
          */
+        @BenignDataRace // See its uses for the explanation of the race
```

Comment:
I think we should explicitly list the races right next to the annotation. This way, if a new race is discovered, we won't look at the annotation and go, "oh yeah, we know about this race, it's benign, no worries."

Reply:
Good idea, fixed